clang/lib/Sema/SemaDecl.cpp

```diff
@@ -8395,10 +8395,11 @@ void Sema::CheckShadow(NamedDecl *D, NamedDecl *ShadowedDecl,
 
   unsigned WarningDiag = diag::warn_decl_shadow;
   SourceLocation CaptureLoc;
-  if (isa<VarDecl>(D) && isa<VarDecl>(ShadowedDecl) && NewDC &&
-      isa<CXXMethodDecl>(NewDC)) {
+  if (isa<VarDecl>(D) && NewDC && isa<CXXMethodDecl>(NewDC)) {
     if (const auto *RD = dyn_cast<CXXRecordDecl>(NewDC->getParent())) {
       if (RD->isLambda() && OldDC->Encloses(NewDC->getLexicalParent())) {
+        if (!isa<VarDecl>(ShadowedDecl))
```

Comment:
Maybe a comment here along the lines of "if it is not VarDecl then it can not shadow" I am not sure if that implies it must be a capture but I think it does.

Reply:
You mean add a comment saying that it is a case when an init-capture references class field?

Reply:
Yeah, just a comment to explain why `ShadownDecl` not being a `VarDecl` matters here.

Reply:
I changed this all, so probably not relevant anymore.